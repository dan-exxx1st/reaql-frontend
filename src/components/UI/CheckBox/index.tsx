import React, { FC } from 'react';
import { ICheckBoxProps } from 'lib/types/components/UI';

import {
    StyledCheckBoxInput,
    StyledCheckBoxLabel,
    StyledCheckBoxSpan,
    StyledCheckBoxText,
    StyledCheckBoxWrapper,
} from './style';

const CheckBox: FC<ICheckBoxProps> = (props) => {
    const { text, onClick } = props;

    return (
        <StyledCheckBoxWrapper>
            <StyledCheckBoxLabel>
                <StyledCheckBoxInput type="checkbox" defaultChecked={false} />
                <StyledCheckBoxSpan onClick={onClick}></StyledCheckBoxSpan>
            </StyledCheckBoxLabel>
            {text ? (
                <StyledCheckBoxText variant="body2">{text}</StyledCheckBoxText>
            ) : null}
        </StyledCheckBoxWrapper>
    );
};

export default CheckBox;
