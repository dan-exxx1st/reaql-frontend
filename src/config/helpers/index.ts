import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
