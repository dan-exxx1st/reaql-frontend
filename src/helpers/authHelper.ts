import { Session, User } from 'lib/graphql/types';
import { useContext } from 'react';
import { UserContext } from './contexts/userContext';

export const CheckAuth = () => {
    const { state } = useContext(UserContext);
    const sessionString = localStorage.getItem('session');
    if (sessionString || state?.user?.email) {
        return true;
    }
    return false;
};

export const getAuthTokens = () => {
    const session = localStorage.getItem('session');
    if (session) {
        const {
            accessToken,
            refreshToken,
        }: { accessToken: string; refreshToken: string } = JSON.parse(session);

        return { accessToken, refreshToken };
    } else {
        return { accessToken: '', refreshToken: '' };
    }
};

export const setSession = (session: {
    refreshToken: string;
    accessToken: string;
}) => {
    const sessionData = {
        accessToken: session.accessToken,
        refreshToken: session.refreshToken,
    };
    const sessionString = JSON.stringify(sessionData);
    localStorage.setItem('session', sessionString);
};

export const setUserAndSession = (payload: {
    user: User;
    session?: Session | null;
}) => {
    const { user, session } = payload;
    if (session) setSession(session);
    localStorage.setItem('userEmail', user.email);
};

export const clearSession = () => {
    localStorage.clear();
    window.location.reload();
};
