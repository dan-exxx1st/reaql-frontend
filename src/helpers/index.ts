import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
