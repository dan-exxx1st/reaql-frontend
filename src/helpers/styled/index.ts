import colorTheme from './colors';
import fontTheme from './fonts';

export { colorTheme, fontTheme };

const effects = {
    formBoxShadow: '5px 5px 15px rgba(37, 37, 37, 0.1)',
};

export default {
    ...colorTheme,
    ...fontTheme,
    ...effects,
};
