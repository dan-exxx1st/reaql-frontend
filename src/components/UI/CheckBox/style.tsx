import styled from 'styled-components';

import { Wrapper, Typography } from '..';

export const StyledCheckBoxWrapper = styled(Wrapper)`
    align-items: center;
`;

export const StyledCheckBoxText = styled(Typography)`
    color: ${({ theme }) => theme.dsc};
`;

export const StyledCheckBoxLabel = styled.label`
    height: 10px;
    display: block;
    position: relative;
    padding-left: 35px;
    margin-bottom: 12px;
    cursor: pointer;
    font-size: 22px;
    user-select: none;

    &:hover input ~ span {
        background-color: ${({ theme }) => theme.wc};
    }

    & > input:checked ~ span:after {
        display: block;
    }

    & > span:after {
        left: 7px;
        top: 3px;
        width: 5px;
        height: 10px;
        border: solid ${({ theme }) => theme.pc};
        border-width: 0 2px 2px 0;
        transform: rotate(45deg);
    }
`;

export const StyledCheckBoxInput = styled.input`
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
`;

export const StyledCheckBoxSpan = styled.span`
    position: absolute;
    top: 0;
    left: 0;
    height: 20px;
    width: 20px;
    border: 1px solid ${({ theme }) => theme.plc};
    border-radius: 2px;

    &:after {
        content: '';
        position: absolute;
        display: none;
    }
`;
