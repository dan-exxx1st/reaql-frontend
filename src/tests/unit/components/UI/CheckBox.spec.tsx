import React from 'react';
import CheckBox from 'components/UI/CheckBox';
import { renderWithTheme } from 'tests/helpers/withTheme';

describe('<CheckBox />', () => {
    it('Check box should render correctly', () => {
        const {
            container: { firstChild },
        } = renderWithTheme(<CheckBox />);
        expect(firstChild).toMatchSnapshot();
    });

    it('Check box with text should render correctly', () => {
        const {
            container: { firstChild },
        } = renderWithTheme(<CheckBox text="Test text" />);
        expect(firstChild).toMatchSnapshot();
    });
});
