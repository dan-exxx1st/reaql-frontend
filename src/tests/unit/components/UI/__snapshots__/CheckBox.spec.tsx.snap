// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> Check box should render correctly 1`] = `
.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  height: 10px;
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover input ~ span {
  background-color: #FAFAFA;
}

.c2 > input:checked ~ span:after {
  display: block;
}

.c2 > span:after {
  left: 7px;
  top: 3px;
  width: 5px;
  height: 10px;
  border: solid #00ADFF;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c3 {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  border: 1px solid #A4E2FF;
  border-radius: 2px;
}

.c4:after {
  content: '';
  position: absolute;
  display: none;
}

<div
  class="c0 c1"
>
  <label
    class="c2"
  >
    <input
      class="c3"
      type="checkbox"
    />
    <span
      class="c4"
    />
  </label>
</div>
`;

exports[`<CheckBox /> Check box with text should render correctly 1`] = `
.c5 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  color: #000000;
  margin: 0;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  color: #555555;
}

.c2 {
  height: 10px;
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover input ~ span {
  background-color: #FAFAFA;
}

.c2 > input:checked ~ span:after {
  display: block;
}

.c2 > span:after {
  left: 7px;
  top: 3px;
  width: 5px;
  height: 10px;
  border: solid #00ADFF;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c3 {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  border: 1px solid #A4E2FF;
  border-radius: 2px;
}

.c4:after {
  content: '';
  position: absolute;
  display: none;
}

<div
  class="c0 c1"
>
  <label
    class="c2"
  >
    <input
      class="c3"
      type="checkbox"
    />
    <span
      class="c4"
    />
  </label>
  <p
    class="c5 c6"
  >
    Test text
  </p>
</div>
`;
